### DISCUSSION TITLE
Workflow failed for get latest template

### DISCUSSION THREAD

**Discussion by clutterwhy** [ObsidianPublisher/obsidian-github-publisher#88] [Q&A — Plugin]
Get lastest release from template workflow run failed.

<img width="539" alt="CleanShot 2023-01-24 at 12 16 01@2x" src="https://user-images.githubusercontent.com/45813757/214211537-e57a990b-85c7-49c5-888f-bf0b22763448.png">

**Comment by Mara-Li** [comment]
Screenshot ? log ?

**Reply by clutterwhy** [reply]
The log

<img width="1039" alt="CleanShot 2023-01-24 at 21 22 36@2x" src="https://user-images.githubusercontent.com/45813757/214305779-00578d7f-5257-49b5-bd91-d3969d677119.png">

**Comment by Mara-Li** [comment]
Okay, after checking, you need to create a github token that allow workflow, and use it in place of the habitual secrets : 

> In order to modify a workflow, a GitHub App, such as the one used for issuing tokens for GitHub Actions, requires the workflow scope. This is so that GitHub Apps you've added to your repository can't access the secrets in your repository without your permission. The token issued for GitHub Actions doesn't have this permission by default.
>
> If you don't need to modify the workflow files, then you can just avoid modifying them and this will go away. If you do need to modify them, you can create an appropriately scoped PAT and store it under a name that doesn't start with GITHUB, say WORKFLOW_TOKEN. > You can then adjust your action to say this:
> `github_token: ${{ secrets.WORKFLOW_TOKEN }}`

**Reply by clutterwhy** [reply]
I have granted permissions to workflows. I read that you need to add permission key in the workflow "auto_update.yml"

<img width="833" alt="CleanShot 2023-01-24 at 22 05 39@2x" src="https://user-images.githubusercontent.com/45813757/214315618-e66fa956-c065-4d53-9c7d-632e4aa101fc.png">


Granular permissions via YAML
You can use the permissions key in the YAML workflow file to modify permissions for the GITHUB_TOKEN for an entire workflow or for individual jobs.

permissions:
  contents: write
  pull-requests: write 
  issues: read
  packages: none

**Reply by Mara-Li** [reply]
Nan.
You need to create a new github token and register it as a secret

**Reply by clutterwhy** [reply]
This?

<img width="778" alt="CleanShot 2023-01-24 at 22 16 17@2x" src="https://user-images.githubusercontent.com/45813757/214318062-f113c6b4-9d76-4ba0-8005-e1aa72bde916.png">

**Reply by Mara-Li** [reply]
https://github.com/settings/tokens

**Reply by Mara-Li** [reply]
Use the old settings because you need to need to set "never" for the expiration

**Reply by clutterwhy** [reply]
<img width="790" alt="CleanShot 2023-01-24 at 22 18 13@2x" src="https://user-images.githubusercontent.com/45813757/214318563-3e96c9e1-3607-4fbc-99c6-cdb56c400654.png">

**Reply by clutterwhy** [reply]
<img width="812" alt="CleanShot 2023-01-24 at 22 19 26@2x" src="https://user-images.githubusercontent.com/45813757/214318778-30bb5c6e-472d-4c9a-8ebd-d2bf5d2a32e4.png">

**Reply by clutterwhy** [reply]
Yes, token is set not to expire. Where can I set it as secret?

**Comment by Mara-Li** [comment]
Hello!
I updated the workflows. You need the 0.0.4 version!

You must do : 
- Create a  `.env` file in `.github` folder.
- With the following value. You need to edit the the WORKFLOW_TYPE to corresponding to your methods (netlify or gh-pages') : 
```
WORKFLOW_TYPE='netlify' or 'gh-pages' (or leave blank : '')
CLEAN=true
IMG_DIR='assets/img'
AUTO_MERGE=false
``` 
- Create the [GH_TOKEN secrets](https://github.com/settings/tokens/new?description=PUBLISHER%20TEMPLATE&scopes=repo,workflow).
- [Register it as a secrets](https://docs.github.com/en/actions/reference/encrypted-secrets#creating-encrypted-secrets-for-a-repository). As I said, the secret must be named `GH_TOKEN`.

**Reply by h7b** [reply]
Thank you. I encountered the same error with the auto-update workflow. With your explanation, today i learned about github workflows, github token, and the use of secrets
